File: chatLLM_MISTRAL_L.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-

import os
import sys
import json
import cgi
import urllib2

# CGI headers
print "Content-Type: text/plain; charset=utf-8"
print ""

try:
    # Lecture POST
    form = cgi.FieldStorage()

    chatBuffer = json.loads(form.getvalue("chatBuffer", "[]"))
    model = json.loads(form.getvalue("model", "\"mistral-large-latest\""))
    temperature = json.loads(form.getvalue("temperature", "0"))
    style = json.loads(form.getvalue("style", "\"\""))
    details = json.loads(form.getvalue("details", "\"\""))

    # Clé API
    api_key = os.environ.get("MISTRAL_API_KEY")
    if not api_key:
        print "Error: Missing MISTRAL_API_KEY"
        sys.exit(1)

    url = "https://api.mistral.ai/v1/chat/completions"

    payload = {
        "model": "mistral-large-latest",
        "messages": chatBuffer,
        "max_tokens": 1000,
        "temperature": temperature
    }

    data = json.dumps(payload)

    request = urllib2.Request(url, data)
    request.add_header("Content-Type", "application/json")
    request.add_header("Authorization", "Bearer " + api_key)

    response = urllib2.urlopen(request, timeout=30)
    response_data = json.loads(response.read())

    # Extraction réponse assistant
    if ("choices" in response_data and
        len(response_data["choices"]) > 0 and
        "message" in response_data["choices"][0] and
        "content" in response_data["choices"][0]["message"]):

        print response_data["choices"][0]["message"]["content"]
    else:
        print "Error: Invalid API response"

except Exception as e:
    print "Error:", str(e)

File: chatLLM_MISTRAL_L.php
<?php
session_start();
$allowedOrigins = [
    "http://Serenity2.local:8888/index.php",
    "https://www.siouxlog.fr/deva2mtl/index.php"
];

$origin = $_SERVER['HTTP_ORIGIN'] ?? '';

if (in_array($origin, $allowedOrigins)) {
    header("Access-Control-Allow-Origin: $origin");
    header("Access-Control-Allow-Credentials: true");
}

header("Access-Control-Allow-Methods: POST");
header("Access-Control-Allow-Headers: Content-Type, X-CSRF-Token");

// Préflight
if ($_SERVER['REQUEST_METHOD'] === 'OPTIONS') {
    http_response_code(204);
    exit;
}
/////////////////////////////////////////////////////////

$apiKey = $_SERVER['MISTRAL_API_KEY'];
$url = 'https://api.mistral.ai/v1/chat/completions';

$headers = array(
    'Content-Type: application/json',
    'Authorization: Bearer ' . $apiKey
);

$messages = json_decode($_POST['chatBuffer'], true);

$model = json_decode($_POST['model'], true);
$temperature = json_decode($_POST['temperature'], true);
$style = json_decode($_POST['style'], true);
$details = json_decode($_POST['details'], true);

/* echo "before";
print_r($chatBuffer);
exit; */

  $data = array(
      'model' =>  "mistral-large-latest", // $model,
      'messages' => $messages,
      'max_tokens' => 1000, // 3000
      'temperature' => $temperature,
      // 'seed' => 12321 // any integer
      // 'top_p' => 0.5
  );
// }

$options = array(
    'http' => array(
        'header' => implode("\r\n", $headers),
        'method' => 'POST',
        'content' => json_encode($data)
    )
);

$context = stream_context_create($options);
$response = file_get_contents($url, false, $context);

// Process the response
if ($response === false) {
    echo 'Error reading response from Mistral API.';
} else {
    $responseData = json_decode($response, true);
    // Access the assistant's reply
    // $assistantReply = end($responseData['choices'])['message']['content'];
    if (isset($responseData['choices'][0]['message']['content'])) {
      $assistantReply = $responseData['choices'][0]['message']['content'];
    }
    echo $assistantReply;
}
?>
